app/src/main/java/com/example/taskyy/ui/viewmodels/ReminderViewModel.kt

```diff
@@ -77,34 +99,106 @@ class ReminderViewModel @Inject constructor(
         }
     }
 
-    private fun formatTimeSelected(time: String) {
-        var timeString = time
-        if (timeString == "0:0") {
-            timeString = ("00:00")
-        }
-        if (timeString == "12:0") {
-            timeString = ("12:00")
+    @OptIn(ExperimentalMaterial3Api::class)
+    private fun formatTimeSelected(time: TimePickerState) {
+
+        val localTime = org.joda.time.LocalTime(time.hour, time.minute)
+        val twelveHourDateFormat =
+            DateTimeFormat.forPattern("hh:mm aa") // 12-hour format with AM/PM
+        val formattedTime = twelveHourDateFormat.print(localTime)
+
+        _state.update { it.copy(selectedTime = formattedTime) }
+    }
+
+    private fun save(title: String, description: String) {
+        val newReminder = Reminder(
+            alarmType = _state.value.formattedReminderType,
+            title = title,
+            description = description,
+            timeInMillis = _state.value.formattedReminderTime,
+            id = Random.nextInt(1..10000),
```

Comment:
The IDs for items are strings in this app. It's quite important that the client generates a unique string ID, so it also works with the backend, even when it gets synced later

Reply:
So how do I get this value? Nothing gets returned so should the user generate this ID?

Reply:
You can generate IDs with `UUID.randomUUID().toString()`

There are many ways to implement a syncing strategy. I personally added a table to the DB where I save which agenda item was locally modified and in which way. E.g. event ID abc was locally created.
Then a WorkManager worker can regularly go through this DB table and sync the pending items and remove them afterwards